background.js

```diff
@@ -17,12 +17,15 @@ installMenuItems.supportsTabContext = true;
 function installBlocker() {
   var list = configs.forceielist.trim().split(/\s+/).filter((aItem) => !!aItem);
   log('force list: ', list);
+  var types = ['main_frame', 'sub_frame'];
+  if (configs.mainFrame)
+    types = ['main_frame'];
   if (list.length > 0 &&
       !browser.webRequest.onBeforeRequest.hasListener(onBeforeRequest))
     browser.webRequest.onBeforeRequest.addListener(
       onBeforeRequest,
       { urls: list,
-        types: ['main_frame', 'sub_frame'] },
+        types: types },
```

Comment:
You just need to put `types` instead of `types: type`. It is a shorthand syntax.
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Object_initializer#%E6%A7%8B%E6%96%87

Reply:
Thanks, It's reasonable. I've fixed.